app/helpers/automated_tests_helper.rb

```diff
@@ -380,6 +474,435 @@ def parse_test_output(repo_assignment_dir, assignment, logfile, data)
+  # before copying to the test server.
+  def self.export_group_repo(group, repo_dir)
+    # Create the automated test repository
+    if !(File.exists?(MarkusConfigurator.markus_config_automated_tests_repository))
+      FileUtils.mkdir(MarkusConfigurator.markus_config_automated_tests_repository)
+    end
+
+    # Delete student's assignment repository if it already exists
+    delete_repo(repo_dir)
+
+    # export
+    return group.repo.export(repo_dir)
+    rescue Exception => e
+      return "#{e.message}"
+  end
+
+  # Delete repository directory
+  def self.delete_repo(repo_dir)
+    # Delete student's assignment repository if it already exists
+    if File.exists?(repo_dir)
+      FileUtils.rm_rf(repo_dir)
+    end
+  end
+
+
+  # Find the list of test scripts to run the test. Return the list of
+  # test scripts in the order specified by seq_num (running order)
+  def self.scripts_to_run(assignment, call_on)
+    # Find all the test scripts of the current assignment
+    all_scripts = TestScript.where(assignment_id: assignment.id)
+
+    list_run_scripts = Array.new
+
+    # If the test run is requested at collection (by Admin or TA),
+    # All of the test scripts should be run.
+    if call_on == "collection"
+      list_run_scripts = all_scripts
+    else
+      # If the test run is requested at submission or upon request,
+      # verify the script is allowed to run.
+      all_scripts.each do |script|
+        if (call_on == "submission") && script.run_on_submission
+          list_run_scripts.insert(list_run_scripts.length, script)
+        elsif (call_on == "request") && script.run_on_request
+          list_run_scripts.insert(list_run_scripts.length, script)
+        end
+      end
+    end
+
+    # sort list_run_scripts using ruby's in place sorting method
+    list_run_scripts.sort_by! {|script| script.seq_num}
+
+    # list_run_scripts should be sorted now. Perform a check here.
+    # Take this out if it causes performance issue.
+    ctr = 0
+    while ctr < list_run_scripts.length - 1
+      if (list_run_scripts[ctr].seq_num) > (list_run_scripts[ctr+1].seq_num)
+        raise "list_run_scripts is not sorted"
+      end
+      ctr = ctr + 1
+    end
+
+    return list_run_scripts
+  end
+
+  # Request an automated test. Ask Resque to enqueue a job.
+  def self.async_test_request(grouping_id, call_on)
+    if files_available?
+      if has_permission?
+        Resque.enqueue(AutomatedTestsHelper, grouping_id, call_on)
+      end
+    end
+  end
+
+
+  # Verify that MarkUs has some files to run the test.
+  # Note: this does not guarantee all required files are presented.
+  # Instead, it checks if there is at least one test script and
+  # source files are successfully exported.
+  def self.files_available?()
+    test_dir = File.join(MarkusConfigurator.markus_config_automated_tests_repository, @assignment.short_identifier)
+    src_dir = @repo_dir
+    assign_dir = @repo_dir + "/" + @assignment.repository_folder
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.